src/utility/common/aglResShaderArchive.cpp

```diff
@@ -0,0 +1,382 @@
+        vertex_shader->uniformVars = modifyBinaryAndNamePtr<GX2UniformVar>(
+            vertex_shader, vertex_shader->uniformVars, vertex_shader->numUniforms);
+        vertex_shader->samplerVars = modifyBinaryAndNamePtr<GX2SamplerVar>(
+            vertex_shader, vertex_shader->samplerVars, vertex_shader->numSamplers);
+        vertex_shader->attribVars = modifyBinaryAndNamePtr<GX2AttribVar>(
+            vertex_shader, vertex_shader->attribVars, vertex_shader->numAttribs);
+
+        vertex_shader->_loopVars = modifyBinaryPtr(vertex_shader, vertex_shader->_loopVars);
+        vertex_shader->shaderPtr = modifyBinaryPtr(vertex_shader, vertex_shader->shaderPtr);
+    } break;
+    case cShaderType_Fragment: {
+        GX2PixelShader* pixel_shader = static_cast<GX2PixelShader*>(getData());
+
+        pixel_shader->uniformBlocks = modifyBinaryAndNamePtr<GX2UniformBlock>(
+            pixel_shader, pixel_shader->uniformBlocks, pixel_shader->numUniformBlocks);
+        pixel_shader->uniformVars = modifyBinaryAndNamePtr<GX2UniformVar>(
+            pixel_shader, pixel_shader->uniformVars, pixel_shader->numUniforms);
+        pixel_shader->samplerVars = modifyBinaryAndNamePtr<GX2SamplerVar>(
+            pixel_shader, pixel_shader->samplerVars, pixel_shader->numSamplers);
+
+        pixel_shader->_loopVars = modifyBinaryPtr(pixel_shader, pixel_shader->_loopVars);
+        pixel_shader->shaderPtr = modifyBinaryPtr(pixel_shader, pixel_shader->shaderPtr);
+    } break;
+    case cShaderType_Geometry: {
+        GX2GeometryShader* geometry_shader = static_cast<GX2GeometryShader*>(getData());
+
+        geometry_shader->uniformBlocks = modifyBinaryAndNamePtr<GX2UniformBlock>(
+            geometry_shader, geometry_shader->uniformBlocks, geometry_shader->numUniformBlocks);
+        geometry_shader->uniformVars = modifyBinaryAndNamePtr<GX2UniformVar>(
+            geometry_shader, geometry_shader->uniformVars, geometry_shader->numUniforms);
+        geometry_shader->samplerVars = modifyBinaryAndNamePtr<GX2SamplerVar>(
+            geometry_shader, geometry_shader->samplerVars, geometry_shader->numSamplers);
+
+        geometry_shader->_loopVars = modifyBinaryPtr(geometry_shader, geometry_shader->_loopVars);
+        geometry_shader->shaderPtr = modifyBinaryPtr(geometry_shader, geometry_shader->shaderPtr);
+        geometry_shader->copyShaderPtr =
+            modifyBinaryPtr(geometry_shader, geometry_shader->copyShaderPtr);
+    } break;
+    }
+#endif  // cafe
+}
+
+const char* ResShaderVariation::getID() const {
+    const char* value = getName() + ref().mNameLen;
+
+    for (s32 i = 0, index = static_cast<s32>(ref().mValueNum);; i++) {
+        while (*value == '\0')
+            value++;
+
+        if (i == index)
+            break;
+
+        while (*value != '\0')
+            value++;
+    }
+
+    return value;
+}
+
+const char* ResShaderVariation::getValue(s32 index) const {
+    SEAD_ASSERT(0 <= index && index < static_cast<int>(ref().mValueNum));
```

Comment:
The spaces here were part of the original assertion message and it wouldn't match anymore now.